cmd/deploy/deploy.go

```diff
@@ -348,3 +373,144 @@ func getProxyHandler(name, token string, clusterConfig *rest.Config) (http.Handl
 	return handler, nil
 
 }
+
+func (*deployCommand) waitUntilDeployed(ctx context.Context, opts *Options) error {
+	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, opts.Name)
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(opts.Timeout)
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("application '%s' didn't finish after %s", opts.Name, opts.Timeout.String())
+		case <-ticker.C:
+			isReady := true
+			dList, err := deployments.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, dep := range dList {
+				if dep.Status.ReadyReplicas != dep.Status.Replicas {
+					isReady = false
+					break
+				}
+			}
+			if !isReady {
+				continue
+			}
+			sfsList, err := statefulsets.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, sfs := range sfsList {
+				if sfs.Status.ReadyReplicas != sfs.Status.Replicas {
+					isReady = false
+					break
+				}
+			}
+
+			if !isReady {
+				continue
+			}
+			return nil
+		}
+	}
+}
+
+func (*deployCommand) showEndpoints(ctx context.Context, opts *Options, spinner *utils.Spinner) error {
+	spinner.Update("retrieving endpoints")
+	pods, err := getDeployPodNames(ctx, opts)
+	if err != nil {
+		return err
+	}
+
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return err
+	}
+	endpoints, err := endpoints.GetEndpointsByPodName(ctx, pods, okteto.Context().Namespace, c)
+	if err != nil {
+		return err
+	}
+
+	svcsName := map[string]bool{}
+	for _, endpoint := range endpoints {
+		svcsName[endpoint.Name] = true
+	}
+
+	iClient, err := ingresses.GetClient(ctx, c)
+	if err != nil {
+		return err
+	}
+	eps, err := iClient.GetEndpointsByEndpointsServices(ctx, okteto.Context().Namespace, svcsName)
+	if err != nil {
+		return err
+	}
+	spinner.Stop()
+	if len(eps) > 0 {
+		sort.Slice(eps, func(i, j int) bool {
+			return len(eps[i]) < len(eps[j])
+		})
+		log.Information("Endpoints available:\n  - %s\n", strings.Join(eps, "\n  - "))
+	}
+	return nil
+}
+
+func getDeployPodNames(ctx context.Context, opts *Options) (map[string]bool, error) {
+	uuids := map[types.UID]bool{}
+	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, opts.Name)
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		log.Infof("error retrieving k8s client: %s", err)
+		return map[string]bool{}, err
+	}
+	dList, err := deployments.List(ctx, okteto.Context().Namespace, labelSelector, c)
+	if err != nil {
+		log.Infof("error retrieving deployments: %s", err)
+		return map[string]bool{}, err
+	}
+	for _, dep := range dList {
+		rs, err := replicasets.GetReplicaSetByDeployment(ctx, &dep, c)
```

Comment:
We are listing all the replicasets in the namespace per deployment to check. Wouldn't it be getter to retrieve the list of all replicasets once and have it in memory? If not, I think we can use the deployment.spec.MatchLabels to filter the number of results here just to the one valid for each deployment